shared/AppInsightsCommon/src/Offline.ts

```diff
@@ -0,0 +1,195 @@
+    // rState is changed by the browser, both via events and when we check the navigator.onLine property
+    let rState: eOfflineValue = eOfflineValue.Online;
+    // ustate is changed by the user calling setOnlineState
+    let uState: eOfflineValue = eOfflineValue.Offline;
+
+    let _evtNamespace = mergeEvtNamespace(createUniqueNamespace("OfflineListener"), parentEvtNamespace);
+
+    try {
+        if (_enableEvents(getWindow())) {
+            _isListening = true;
+        }
+        
+        if (_document) {
+            // Also attach to the document.body or document
+            let target:any = _document.body || _document;
+
+            if (target.ononline) {
+                if (_enableEvents(target)) {
+                    _isListening = true;
+                }
+            }
+        }
+
+        if (_isListening) {
+            // We are listening to events so lets set the current status rather than assuming we are online #1538
+            if (_navigator && !isNullOrUndefined(_navigator.onLine)) { // navigator.onLine is undefined in react-native
+                _currentState = _navigator.onLine;
+            }
+        }
+    } catch (e) {
+        // this makes react-native less angry
+        _isListening = false;
+    }
+
+    function _enableEvents(target: any): boolean {
+        let enabled = false;
+        if (target) {
+            enabled = eventOn(target, "online", _setOnline, _evtNamespace);
+            if (enabled) {
+                eventOn(target, "offline", _setOffline, _evtNamespace);
+            }
+        }
+        return enabled;
+    }
+
+    function _isOnline(){
+        return _currentState;
+    }
+
+    function calCurrentState(){
+        if (uState === eOfflineValue.Offline || rState === eOfflineValue.Offline){
+            return false;
+        }
+        return true; // if both unknown, then we assume the network is good
+    }
+
+    function listnerNoticeCheck(){
+        // we were offline and are now online or we were online and now offline
+        let newState = calCurrentState();
+        if (_currentState !== newState) {
+            _currentState = newState; // use the resolved state to update
+            // send all the callbacks with the current state
+            arrForEach(listenerList, (callback: OfflineCallback) => {
+                let offlineState: IOfflineState = {
+                    isOnline: _currentState,
+                    rState: rState,
+                    uState: uState
+                };
+                callback(offlineState);
+            });
+        }
+    }
+
+    function setOnlineState (uState: eOfflineValue){
+        this.uState = uState;
+        listnerNoticeCheck();
+    }
+
+    function _setOnline() {
+        rState = eOfflineValue.Online;
+        listnerNoticeCheck();
+    }
+
+    function _setOffline() {
+        rState = eOfflineValue.Offline;
+        listnerNoticeCheck();
+    }
+
+    function _isNavOnline(): boolean {
```

Comment:
This looks like it was the old "isOnline()" which is only handling the now `rState`

Reply:
yes, it is the old isOnline, and now it is for rState init use only